src/metabase/query_processor/middleware/metrics.clj

```diff
@@ -263,7 +263,7 @@
                          (update stage-or-join :stages #(adjust-metric-stages query path %)))))]
           (u/prog1
             (update query :stages #(adjust-metric-stages query nil %))
-            (when-let [metric (find-first-metric <>)]
+            (when-let [metric (find-first-metric (:stages <>))]
```

Comment:
Should the call to `find-first-metric` in the `if-not` above also look only at `:stages`?

Reply:
Added it there as well. (My understanding is that this is only to do less useless work, so I haven't added a test for this.)